src/signature/policy/policy_requirement.rs

```diff
@@ -40,14 +48,49 @@ pub enum PolicyReqType {
     // Refer to issue: https://github.com/confidential-containers/image-rs/issues/7
 }
 
+/// Copy cosign parameters struct from mechansim/cosign/mod.rs when image-rs isn't
+/// built with the cosign module
+#[cfg(not(feature = "cosign"))]
+#[derive(Deserialize, Debug, Eq, PartialEq, Serialize, Default)]
+pub struct CosignParameters {
```

Comment:
IMOP, if we decided not to support cosign in a feature, we do not need to define extra cosign-related structures to make deserialize successful and tell the user `image-rs was built without support for cosign image signing`, but directly an error that can be raised by `serde` when deserializing the `policy.json`. I'm considering duplicated code her, but still not sure about this. What is your opinion?

Reply:
The error I got in the `signature_verification_one_kbc::case_1` test on a s390x machine when the `CosignParameters` struct is not included in `policy_requirement.rs` and the `Cosign(CosignParameters)` enumeration is conditionally taken out of the `PolicyReqType` enum when building on s390x is 
```
Signing scheme: None, Err(Security validate failed: unknown variant `sigstoreSigned`, expected one of `insecureAcceptAnything`, `reject`, `signedBy`
```

which as an error doesn't indicate that image-rs has been built without support for cosign image signing.

It may be possible to remove the duplicated `CosignParameters` struct in `policy_requirement.rs` if the `CosignParameters` struct can be imported from `cosign/mod.rs` without also including any part of the sigstore_rs create.

Reply:
Including `cosign::CosignParameters` on s390x from in `policy_requirement.rs` requires removing `#[cfg(feature = "cosign")]` from

```
#[cfg(feature = "cosign")]
pub mod cosign;
```

which requires we can import from the sigstore_rs crate:

```
   Compiling image-rs v0.1.0 (/root/image-rs)
error[E0433]: failed to resolve: use of undeclared crate or module `sigstore_rs`
  --> src/signature/mechanism/cosign/mod.rs:14:5
   |
14 | use sigstore_rs::{
   |     ^^^^^^^^^^^ use of undeclared crate or module `sigstore_rs`
```

Based on that, I think the options are to duplicate `CosignParameters` in `policy_requirement.rs` and have the error that tells the user that image-rs was built without support for cosign, and have the duplicated code, or have the error from failed deserialisation.

I'll double check what happens in the test when the error comes from the failed deserialisation.